test/StakeManager.t.sol

```diff
@@ -601,36 +644,46 @@ contract ExecuteAccountTest is StakeManagerTest {
             }
         }
 
-        for (uint256 i = 0; i < 100; i++) {
+        for (uint256 i = 0; i < 5; i++) {
             deal(stakeToken, address(stakeManager), 100 ether);
             vm.warp(stakeManager.epochEnd());
             stakeManager.executeEpoch();
             for (uint256 j = 0; j < userVaults.length; j++) {
-                (address rewardAddress,,, uint256 totalMPBefore, uint256 lastMintBefore,, uint256 epochBefore) =
-                    stakeManager.accounts(address(userVaults[j]));
+                (
+                    address rewardAddress,
+                    ,
+                    ,
+                    uint256 totalMPBefore,
+                    uint256 lastMintBefore,
+                    ,
+                    uint256 epochBefore,
+                ) = stakeManager.accounts(address(userVaults[j]));
                 uint256 rewardsBefore = ERC20(stakeToken).balanceOf(rewardAddress);
 
                 stakeManager.executeAccount(address(userVaults[j]), epochBefore + 1);
-                (,,, uint256 totalMP, uint256 lastMint,, uint256 epoch) = stakeManager.accounts(address(userVaults[j]));
+                (,,, uint256 totalMP, uint256 lastMint,, uint256 epoch,) =
+                    stakeManager.accounts(address(userVaults[j]));
                 uint256 rewards = ERC20(stakeToken).balanceOf(rewardAddress);
                 assertEq(lastMint, lastMintBefore + stakeManager.EPOCH_SIZE(), "must increaase lastMint");
                 assertEq(epoch, epochBefore + 1, "must increase epoch");
-                assertEq(totalMP, totalMPBefore, "must NOT increase MPs");
+                // MPs will still be minted in mpLimitEpoch + 1 when accounts
+                // started staking at any point *inside* of an epoch, so we
+                // only perform the assert below one epoch after *that*
+                if (i > 0) {
+                    assertEq(totalMP, totalMPBefore, "must NOT increase MPs");
+                }
```

Comment:
@3esmit this is a fix I've introduced as it would otherwise immediately have a failing assertion.
Reason being is that we always have that 1 epoch after maxlimit epoch in which MPs will be minted, **if** there's accounts that started staking after the start time of any given epoch (which is likely 99% the case in reality).

So we have to check for greater `0` here to perform the assertion only after we've really minted the last MPs.

Reply:
Is not always, I am not sure about that. And in some cases (really small staking values), can be 2 epochs.

It should use the same calculations as in the smart contract, or better, use precalculated values and precalculated static checks.